src/setting.py

```diff
@@ -127,7 +131,11 @@ class AppSetting:
 
     def __init__(self, **kwargs):
         self.__port = kwargs.get('port') or AppSetting.PORT
-        self.__data_dir = self.__compute_dir(kwargs.get('data_dir'), AppSetting.default_data_dir)
+        self.__global_dir = self.__compute_dir(kwargs.get('global_dir'), AppSetting.default_global_dir, 0o777)
+        self.__data_dir = self.__compute_dir(kwargs.get('data_dir'),
+                                             os.path.join(self.global_dir, AppSetting.default_data_dir))
+        self.__config_dir = self.__compute_dir(kwargs.get('config_dir'),
+                                               os.path.join(self.global_dir, AppSetting.default_config_dir))
```

Comment:
Use `self.__global_dir`

For production also convention should be same. If we give commnad: `-g /data/rubix-point-server -d data -c config`, it should result:

- data directory: `/data/rubix-point-server/data`
- config directory: `/data/rubix-point-server/config`